server/src/main/scala/org/http4s/server/middleware/authentication/package.scala

```diff
@@ -10,14 +10,16 @@ package object authentication {
   type Realm = String
   type User = String
   type ClientPasssword = String
-  // A function mapping (realm, username) to password, None if no password
-  // exists for that (realm, username) pair.
+  /**
+    * A function mapping (realm, username) to password, None if no password
+    * exists for that (realm, username) pair.
+    */
   type AuthenticationStore = (Realm, User) => Task[Option[String]]
   /**
     *  Validates a plaintext password (presumably by comparing it to a hashed value).
-    *  Some(true) means the password given is valid.
+    *  A Some value indicates success; None indicates the password failed to validate.
     */
-  type ValidatePassword = (Realm, User, ClientPasssword) => Task[Option[Boolean]]
+  type AuthenticationValidator = (Realm, User, ClientPasssword) => Task[Option[Unit]]
```

Comment:
`Option[Unit]` feels awkward to me.  I think this is either a `Boolean` (which is often a smell) or an ADT.  Might it make sense for this to just return the `AuthReply`?  Would involve making that public.